sdks/go/pkg/beam/runners/dataflow/translate.go

```diff
@@ -351,6 +363,45 @@ func translateEdge(edge *graph.MultiEdge) (string, properties, error) {
 			UserName: buildName(edge.Scope(), "flatten"), // TODO: user-defined
 		}, nil
 
+	case graph.External:
```

Comment:
Ack. The Dataflow translation won't likely change further and any other interpretation of the external payloads would have to be supported by the underlying engine. Instead of adding a stateful registration to an otherwise pure translation, substitutions in the proto pipeline would be a preferable way to go long term.